src/application/components/MainPanel.tsx

```diff
@@ -66,17 +67,20 @@ export function MainPanelPane({ pane, pdfViewerRef }: MainPanelPaneProps & MainP
   }));
 
   const selectFileInPane = useSetAtom(selectEditorInPaneAtom);
-  const closeFileInPane = useSetAtom(closeEditorFromPaneAtom);
   const focusPane = useSetAtom(focusPaneAtom);
 
+  const handleCloseClick = (paneEditorId: PaneEditorId) => {
+    emitEvent(RefStudioEvents.editors.close, paneEditorId);
+  };
```

Comment:
nit: we can make this "inline" and even maybe use `emitEvent` directly in the `onCloseClick` handler bellow.